assets/src/block-editor/setupBlockEditorValidation.js

```diff
@@ -0,0 +1,182 @@
+ * It subscribes to changes in the editor and performs validation checks on title, featured image, and blocks.
+ */
+export const setupBlockEditorValidation = () => {
+  subscribe(() => {
+    const {getEditedPostAttribute, getCurrentPostType, getEditedPostContent} = select('core/editor');
+    const {getBlocks} = select('core/block-editor');
+
+    // Get the necessary data from the editor state
+    const title = getEditedPostAttribute('title');
+    const featuredImage = getEditedPostAttribute('featured_media');
+    const postType = getCurrentPostType();
+    const postContent = getEditedPostContent();
+    const blocks = getBlocks();
+    const currentMessages = [];
+
+    // Validation for title (must not be empty)
+    const invalidTitle = !title || title.trim().length <= 0;
+    if (invalidTitle) {
+      currentMessages.push('Title is required.');
+    }
+
+    // Check if the post requires a featured image and if it's missing
+    const hasImageInContent = /<img.+wp-image-(\d+).*>/i.test(postContent);
+    const needsFeaturedImage = POST_TYPES_WITH_REQUIRED_FEATURED_IMAGE.includes(postType) &&
+      !featuredImage &&
+      !hasImageInContent;
+
+    if (needsFeaturedImage) {
+      currentMessages.push('Featured image is required.');
+    }
+
+    // Get an array of Topic Link blocks without background image
+    const topicLinkBlocksNeedsImage = blocks
+      .filter(block => block.name === 'planet4-blocks/topic-link')
+      .filter(block => !block.attributes.imageUrl);
+
+    // If there are Topic Link blocks without background image, push an error message
+    if (topicLinkBlocksNeedsImage.length > 0) {
+      currentMessages.push('Background image in the Topic Link block is required.');
+    }
+
+    const invalidBlocks = blocks.reduce((invalidBlocksArray, block) => {
+      // Normally `blocks` contains a valid list of blocks, however it can happen that one of them is `null` in rare
+      // cases. It happened to me once while running with WordPress 5.8.1 and undoing multiple edits. This made the
+      // editor crash while it's trying to access `block.name`.
+      if (!block) {
+        return; // eslint-disable-line array-callback-return
+      }
+      const validations = blockValidations[block.name] || {};
+
+      const results = Object.entries(validations).reduce((resultsArray, [attrName, validate]) => {
+        const value = block.attributes[attrName];
+        const result = validate(value);
+
+        if (!result.isValid) {
+          resultsArray.push(result);
+        }
+
+        return resultsArray;
+      }, []);
+
+      invalidBlocksArray.push(...results);
+
+      return invalidBlocksArray;
+    }, []);
+
+    // Push invalid block validation messages to currentMessages
+    invalidBlocks.forEach(block => currentMessages.push(...block.messages));
+
+    // Determine if the post is currently valid (no validation errors)
+    const currentlyValid =
+      (0 === invalidBlocks.length) &&
```

Comment:
You can also check like this
```suggestion
      (!invalidBlocks.length) &&
```

Reply:
Done!